### PR TITLE
Check index DDL 

### PR DESCRIPTION
**Pull request by Moonlit-Sailor**
## What changes were proposed in this pull request?
Add DDL for index checking: `check oindex on table`

currently, only support checking OAP meta file's existence for a specified table
Output of the checking results:
 for partition directory which has no .oap.meta file, output a row:
**Meta file not found in partition: partition directory**
## How was this patch tested?
unit tests

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala [hunk 1]
```diff
@@ -496,3 +498,51 @@ case class OapShowIndex(table: TableIdentifier, relationName: String)
...
+  def checkOapMetaFile(fs: FileSystem,
+                       partitionDirs: Seq[PartitionDirectory]): (Seq[Path], Seq[Row]) = {
+    require(null ne fs, "file system should not be null!")
+    val (partWithMeta, partWithoutMeta) =
+      partitionDirs.map(_.files.head.getPath.getParent)
+        .partition(partitionDir => fs.exists(new Path(partitionDir, OapFileFormat.OAP_META_FILE)))
+
+    (partWithMeta,
+      partWithoutMeta.map(partitionPath =>
+        Row(s"Meta file not found in partition: ${partitionPath.toUri.getPath}")))
```
Comment (linhong-intel) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: About the return values, here is my suggestion:
Option 1:
Only return `Seq[Row]` to report the check result.
Option 2:
return (Seq[Path], Seq[Path]) to report the` partWithMeta` and `partWithoutMeta`

I prefer one function only does one job. return both partWithMeta and check result will make thing complicated

Comment (Moonlit-Sailor) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: ok, this method returns (Seq[Path], Seq[Path]) of partition directories with meta and without meta for subsequent checking.

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala [hunk 2]
```diff
@@ -496,3 +498,51 @@ case class OapShowIndex(table: TableIdentifier, relationName: String)
...
+      EliminateSubqueryAliases(sparkSession.sessionState.catalog.lookupRelation(table)) match {
+        case r: SimpleCatalogRelation => new FindDataSourceTable(sparkSession)(r)
+        case other => other
+      }
+
+    val (fileCatalog, dataSchema) = relation match {
+      case LogicalRelation(HadoopFsRelation(f, _, s, _, _, _), _, id) =>
+        (f, s)
+      case other =>
+        throw new OapException(s"We don't support index listing for ${other.simpleString}")
```
[the remark this answers is not in the document]
Comment (Moonlit-Sailor) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: done

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala [hunk 3]
```diff
@@ -496,3 +498,51 @@ case class OapShowIndex(table: TableIdentifier, relationName: String)
     })
   }
 }
+
+case class OapCheckIndex(table: TableIdentifier, tableName: String)
+  extends RunnableCommand with Logging {
+  override val output: Seq[Attribute] =
+    AttributeReference("Analysis Result", StringType, nullable = false)() :: Nil
+  def checkOapMetaFile(fs: FileSystem,
```
[the remark this answers is not in the document]
Comment (Moonlit-Sailor) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: done

File: src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 4]
```diff
@@ -714,7 +715,7 @@ nonReserved
     | DESCRIBE | DROP | EXISTS | FALSE | FOR | GROUP | IN | INSERT | INTO | IS |LIKE
     | NULL | ORDER | OUTER | TABLE | TRUE | WITH | RLIKE
     | AND | CASE | CAST | DISTINCT | DIV | ELSE | END | FUNCTION | INTERVAL | MACRO | OR | STRATIFY | THEN
-    | UNBOUNDED | WHEN
+    | UNBOUNDED | WHEN | CHECK
```
[the remark this answers is not in the document]
Comment (Moonlit-Sailor) [reply src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: ok, put it at the bottom.

**Comment by linhong-intel** [comment]
LGTM

**Comment by chenghao-intel** [comment]
LGTM.

**Comment by chenghao-intel** [comment]
thank you @Moonlit-Sailor please continue to the index / data consistency checkings, and let's make it smaller PRs one by one.

File: src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 5]
```diff
@@ -804,6 +806,7 @@ USE: 'USE';
 PARTITIONS: 'PARTITIONS';
 FUNCTIONS: 'FUNCTIONS';
 DROP: 'DROP';
+CHECK: 'CHECK';
```
Comment (adrian-wang) [inline src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: better move this near other oap-added words.

**Pull request merged**